### ISSUE TITLE
Minified lib returns an empty string

### ISSUE THREAD

**Issue by bricss** [wycats/handlebars.js#1129]
Hi there,

I have found that latest version of minified [handlebars.**min.js**](http://builds.handlebarsjs.com.s3.amazonaws.com/handlebars.min-v4.0.4.js) surprisingly returns an empty string every time on each compile operation.

For example:
```javascript
var tpl = Handlebars.compile('{{foo}}');
var output = tpl({foo: 'bar'});
console.log(output === ''); // true
```

**Comment by kpdecker** [comment]
This is unexpected. Could you create a fiddle demonstrating the issue? A
template is provided in the contributing.md document. Also what
environments are you seeing this issue in?

**Comment by bricss** [comment]
I found that problem when start preparing production version of our web app. We are using handlebars to render course quizzes to our users. And when we have updated to latest version quizzes stop to appear in html courses. After half an hour of debugging I found where was a problem when changed minified file to ordinary.

Later today I will add example.

**Comment by kpdecker** [comment]
By environment, I mean what browser or other javascript engine.

**Comment by bricss** [comment]
Every, chrome 46, firefox 42, ie10-11-edge, opera, safari and ios webview.

**Comment by bricss** [comment]
Proof of concept -> http://jsfiddle.net/pmtytqoa/1/

**Comment by kpdecker** [comment]
Not sure what went wrong but I was able to write some tests that reproduce this failure. It appears to be specific to the full library when minimized. The runtime library works properly, at least when run in the node test environment.

**Comment by kpdecker** [comment]
This is somehow caused by https://github.com/gruntjs/grunt-contrib-uglify/issues/366. Leaving the comments in place caused something to break after minimization in a way that is not clear to me.

**Comment by kpdecker** [comment]
Released in 4.0.5
